spec/employee_data_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'csv'
+require 'rspec'
+require 'byebug'
+
+RSpec.describe "employee data" do
+  def csv_data
+    @array ||= []
+    if @array.length.zero?
+      CSV.foreach(("employee_data.csv"), headers: true, col_sep: ",") do |row|
+        @array << row
+      end
+    end
+    @array
+  end
+
+# Identify employees who joined before 2020 and have a salary above $75,000
+  context "employee information" do
+    def employee_info
+      data = []
+        csv_data.each do |row|
+          if row["JoinDate"] < '2020-01-01' && row["Salary"].to_i > 75000
+            data << {name: row["Name"],
+            join_date: row["JoinDate"],
+            salary: row["Salary"]}
+          end
+        end
+      data
+    end 
+
+    it "Identify employee's joined date & salary" do 
+      expect(employee_info).to eq([])
+    end
+  end
+
+  #================================================================================
+  # Determine the average salary of each department.
+  context "Average salary" do
+    def avergae_salary
+       salary_department = {}
+        csv_data.each do |row|
+          department = row['Department']
+          salary = row['Salary'].to_i
+          if salary_department[department]
+            salary_department[department] << salary
+          elsif
+            salary_department[department] = [salary]
+          end
```

Comment:
Two things here:

1. `elsif` needs to have a condition or use `else` there. However check point 2
2. There is a better way we did in the previous examples. I would do that.